test/TestNestedJinjaRule.py

```diff
@@ -0,0 +1,190 @@
+      {{ 3 - 1 }}
+      }} }}
+''')
+
+FAIL_TASK_W_5LN = PlayFile('playbook.yml', u'''
+- name: five-level wild nesting
+  set_fact:
+    var_three_wld: "{{ {{ {{ {{ {{ 234 }} }} }} }} }}"
+''')
+
+FAIL_TASK_W_5LN_M = PlayFile('playbook.yml', u'''
+- name: five-level wild multiline nesting
+  set_fact:
+    var_three_wld_ml: >
+      {{
+      {{
+      {{
+      {{
+      {{ 234 }}
+      }}
+      }}
+      }}
+      }}
+''')
+
+SUCCESS_TASK_P = PlayFile('playbook.yml', u'''
+- name: proper non-nested example
+  set_fact:
+    var_one: "number for 'one' is {{ 2 * 1 }}"
+''')
+
+SUCCESS_TASK_P_M = PlayFile('playbook.yml', u'''
+- name: proper multiline non-nested example
+  set_fact:
+    var_one_ml: >
+      number for 'one' is {{
+      2 * 1 }}
+''')
+
+SUCCESS_TASK_2P = PlayFile('playbook.yml', u'''
+- name: proper nesting far from each other
+  set_fact:
+    var_two: "number for 'two' is {{ 2 * 1 }} and number for 'three' is {{ 4 - 1 }}"
+''')
+
+SUCCESS_TASK_2P_M = PlayFile('playbook.yml', u'''
+- name: proper multiline nesting far from each other
+  set_fact:
+    var_two_ml: >
+      number for 'two' is {{ 2 * 1
+      }} and number for 'three' is {{
+      4 - 1 }}
+''')
+
+SUCCESS_TASK_C_2P = PlayFile('playbook.yml', u'''
+- name: proper nesting close to each other
+  set_fact:
+    var_three: "number for 'ten' is {{ 2 - 1 }}{{ 3 - 3 }}"
+''')
+
+SUCCESS_TASK_C_2P_M = PlayFile('playbook.yml', u'''
+- name: proper multiline nesting close to each other
+  set_fact:
+    var_three_ml: >
+      number for 'ten' is {{
+      2 - 1
+      }}{{ 3 - 3 }}
+''')
+
+@pytest.fixture
+def play_file_path(tmp_path):
+    p = tmp_path / 'playbook.yml'
+    return str(p)
+
+
+@pytest.fixture
+def rules():
+    rulesdir = os.path.join('lib', 'ansiblelint', 'rules')
+    return RulesCollection([rulesdir])
+
+
+@pytest.fixture
+def runner(play_file_path, rules):
+    return Runner(rules, play_file_path, [], [], [])
+
+
+@pytest.fixture
+def play_files(tmp_path, request):
+    if request.param is None:
+        return
+    for play_file in request.param:
+        p = tmp_path / play_file.name
+        p.write_text(play_file.content)
+
+
+@pytest.mark.parametrize(
+    'play_files',
+    [pytest.param(
+        [
+            FAIL_TASK_1LN,
+            FAIL_TASK_1LN_M,
+            FAIL_TASK_2LN,
+            FAIL_TASK_2LN_M,
+            FAIL_TASK_W_5LN,
+            FAIL_TASK_W_5LN_M
+        ],
+        id='file includes nested jinja pattern'
+    )],
+    indirect=['play_files']
+)
+def test_including_nested_jinja(runner, play_files):
+    results = str(runner.run())
+    assert 'nested jinja' in results
+
+
+@pytest.mark.parametrize(
+    'play_files',
+    [pytest.param(
+        [
+            SUCCESS_TASK_P,
+            SUCCESS_TASK_P_M,
+            SUCCESS_TASK_2P,
+            SUCCESS_TASK_2P_M,
+            SUCCESS_TASK_C_2P,
+            SUCCESS_TASK_C_2P_M
+        ],
+        id='file does not include nested jinja pattern'
+    )],
```

Comment:
You created one param with a nested list of tasks. But you need multiple params with each of them having its own id so that it will generate a new test per param. Once done, each parametrized subtest will be visible in the pytest output.